src/battery/zendure/Provider.cpp

```diff
@@ -0,0 +1,878 @@
+            );
+    log("Subscribed to '%s' for timesync requests", _topicTimesync.c_str());
+
+#ifndef ZENDURE_NO_REDUCED_UPDATE
+    // subscribe for read messages
+    _topicReadReply = _baseTopic + "properties/read/reply";
+    MqttSettings.subscribe(_topicReadReply, 0/*QoS*/,
+            std::bind(&Provider::onMqttMessageReport,
+                this, std::placeholders::_1, std::placeholders::_2,
+                std::placeholders::_3, std::placeholders::_4,
+                std::placeholders::_5, std::placeholders::_6)
+            );
+    log("Subscribed to '%s' for status readings\r\n", _topicReadReply.c_str());
+
+    _rateUpdateMs       = min(static_cast<uint32_t>(config.Battery.ZendurePollingInterval * 100), 10U * 1000);
+    _nextUpdate         = 0;
+#endif
+
+    _rateFullUpdateMs   = config.Battery.Zendure.PollingInterval * 1000;
+    _nextFullUpdate     = 0;
+    _rateTimesyncMs     = ZENDURE_SECONDS_TIMESYNC * 1000;
+    _nextTimesync       = 0;
+    _rateSunCalcMs      = ZENDURE_SECONDS_SUNPOSITION * 1000;
+    _nextSunCalc        = millis() + _rateSunCalcMs / 2;
+
+    // pre-generate the settings request
+    JsonDocument root;
+    JsonVariant prop = root[ZENDURE_REPORT_PROPERTIES].to<JsonObject>();
+    prop[ZENDURE_REPORT_PV_BRAND] = 1; // means Hoymiles
+    prop[ZENDURE_REPORT_PV_AUTO_MODEL] = 0; // we did static setup
+    prop[ZENDURE_REPORT_AUTO_RECOVER] = static_cast<uint8_t>(config.Battery.Zendure.BypassMode == static_cast<uint8_t>(BypassMode::Automatic));
+    prop[ZENDURE_REPORT_AUTO_SHUTDOWN] = static_cast<uint8_t>(config.Battery.Zendure.AutoShutdown);
+    prop[ZENDURE_REPORT_BUZZER_SWITCH] = 0; // disable, as it is anoying
+    prop[ZENDURE_REPORT_BYPASS_MODE] = config.Battery.Zendure.BypassMode;
+    prop[ZENDURE_REPORT_SMART_MODE] = 0; // should be disabled
+    serializeJson(root, _payloadSettings);
+
+    // pre-generate the full update request
+    root.clear();
+    JsonArray array = root[ZENDURE_REPORT_PROPERTIES].to<JsonArray>();
+    array.add("getAll");
+    array.add("getInfo");
+    serializeJson(root, _payloadFullUpdate);
+
+#ifndef ZENDURE_NO_REDUCED_UPDATE
+    // pre-generate the partitial update request
+    root.clear();
+    array = root[ZENDURE_REPORT_PROPERTIES].to<JsonArray>();
+    array.add(ZENDURE_REPORT_MIN_SOC);
+    array.add(ZENDURE_REPORT_MAX_SOC);
+    array.add(ZENDURE_REPORT_INPUT_LIMIT);
+    array.add(ZENDURE_REPORT_OUTPUT_LIMIT);
+    array.add(ZENDURE_REPORT_INVERSE_MAX_POWER);
+    array.add(ZENDURE_REPORT_BATTERY_STATE);
+    array.add(ZENDURE_REPORT_HEAT_STATE);
+    array.add(ZENDURE_REPORT_AUTO_SHUTDOWN);
+    array.add(ZENDURE_REPORT_BUZZER_SWITCH);
+    array.add(ZENDURE_REPORT_REMAIN_OUT_TIME);
+    array.add(ZENDURE_REPORT_REMAIN_IN_TIME);
+    serializeJson(root, _payloadUpdate);
+#endif
+
+    // initial setup
+    if (!config.Battery.Zendure.ChargeThroughEnable) {
+        setChargeThrough(false);
+    }
```

Comment:
Should this look like this instead to cover both cases enabled and disabled?
```suggestion
    setChargeThrough(config.Battery.Zendure.ChargeThroughEnable);
```

Reply:
Nope, this aborts charge through when it got disabled by the user. Passing `true` to it, would explicitly start it (what should only be done by the timer or when received via MQTT)